Add representative image to Collection objects

paclough opened the pull request:
Add create and update methods to the CollectionController to call
add_profile_image if an image is uploaded. Also add an index method to
redirect to the "My Collections" catalog view in order to display
Collection thumbnails (rather than modify the Blacklight method).
Include Hydra::Derivatives with the Collection model and add the
necessary datastreams, accessors and methods; used Person model as a
template. Add upload field to the Collection form. Add the
thumbnail display to the Collection show view and add another check
to the thumbnail partial (check for the presence of the 'thumbnail'
datastream). Add a new test to the collection feature spec.

paclough committed:
Add representative image to Collection objects

Add create and update methods to the CollectionController to call
add_profile_image if an image is uploaded. Also add an index method to
redirect to the "My Collections" catalog view in order to display
Collection thumbnails (rather than modify the Blacklight method).
Include Hydra::Derivatives with the Collection model and add the
necessary datastreams, accessors and methods; used Person model as a
template. Add upload field to the Collection form. Add the
thumbnail display to the Collection show view and add another check
to the thumbnail partial (check for the presence of the 'thumbnail'
datastream). Add a new test to the collection feature spec.

scherztc commented on app/views/catalog/_index_partials/_thumbnail_display.html.erb:
```diff
@@ -1,6 +1,6 @@
 <%- if document.respond_to?(:representative_image_url) && document.representative_image_url.present? %>
   <%= image_tag document.representative_image_url, class: "canonical-image"  %>
-<%- elsif document.respond_to?(:representative) && document.representative.present? %>
+<%- elsif document.respond_to?(:representative) && document.representative.present? && document.thumbnail.present?%>
```
I get an undefined method `thumbnail' error when trying to view this page.  This is where Rajesh and I were discussing the use of a thumbnail validation method and the cost of said method.

paclough committed:
Move functionality to model

Reconfigured the controller to just set a model variable during
create or update, then the model uses a before_save callback to
access the var. Updated spec to verify redirect from index to
the collections facet of the catalog.
